src/main/java/edu/ucla/library/services/utility/PropsWriter.java

```diff
@@ -0,0 +1,165 @@
+
+package edu.ucla.library.services.utility;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.util.Properties;
+
+public final class PropsWriter {
+
+    /**
+     * Properties file used as source for output file.
+    */
+    private static final Properties PROPS = new Properties();
+
+    /**
+     * Expected number of parameters passed to class.
+    */
+    private static final int ARGS_COUNT = 3;
+
+    /**
+     * Delimiter character for value keys in template file.
+    */
+    private static final String DELIMITER = "$";
+
+    /**
+     * Private constructor for PropsWriter class.
+    */
+    private PropsWriter() {
+        super();
+    }
+
+    /**
+     * Main method for command-line execution.
+     *
+     * @param args array of parameters
+    */
+    @SuppressWarnings("uncommentedmain")
+    public static void main(final String[] args) {
+        verifyArgs(args);
+        loadProps(args[0]);
+        readWrite(args[1], args[2]);
+    }
+
+    /**
+     * Public method for testing access or plugging into other classes.
+     *
+     * @param aDefaults default values fed to PROPS
+     * @param aTemplate template for the output file
+     * @param aOutput properties file built by app from aDefaults and aTemplate
+    */
+    public static void buildProperties(final String aDefaults,
+        final String aTemplate, final String aOutput) {
+        verifyArgs(new String[] {aDefaults, aTemplate, aOutput});
+        loadProps(aDefaults);
+        readWrite(aTemplate, aOutput);
+    }
+
+    /**
+     * Check that parameters are valid.
+     *
+     * @param aArray array of file names
+    */
+    public static void verifyArgs(final String[] aArray) {
+        if (aArray.length != ARGS_COUNT) {
+            System.err.println(
```

Comment:
Might be nice to use the MessageCodes/Logger for this so it can be internationalized, but we can add this later.